### DISCUSSION TITLE
Help with clonotype analysis of BCR cDNA library

### DISCUSSION THREAD

**Discussion by liyunjian** [milaboratory/mixcr#1673] [Q&A] [answered]
I always see a high percentage of Unassigned alignments in clonotype assembly in the output of MIXCR.
The output clones are around 8000. I want to ask if this is normal.
This is a BCR cDNA library sequenced by Illumina Novaseq6000 SP PE250。

The command I used is as follows:
mixcr analyze generic-amplicon-with-umi ./oneUMI/JLW2KL_R1A.fastq ./oneUMI/JLW2KL_R2A.fastq ./oneUMI/JLW2KL \
    -Xmx50g \
    -Massemble.cloneAssemblerParameters.addReadsCountOnClustering=true \
    --species hsa \
    --rna \
    --tag-pattern "^(UMI:N{8})(R1:*)\^(R2:*)" \
    --floating-left-alignment-boundary \
    --floating-right-alignment-boundary J \
    --assemble-clonotypes-by "VDJRegion" \
    --impute-germline-on-export \
    --threads 32

Here is the report I output:
·····································································································
Analysis date: Sun May 26 16:14:09 CST 2024
Input file(s): ./oneUMI/JLW2H_R1A.fastq,./oneUMI/JLW2H_R2A.fastq
Output file(s): ./oneUMI/JLW2H/JLW2H.vdjca
Version: 4.6.0; built=Sun Dec 10 03:48:42 CST 2023; rev=c9fafa41fe; lib=repseqio.v4.0
Command line arguments: align --report ./oneUMI/JLW2H/JLW2H.align.report.txt --json-report ./oneUMI/JLW2H/JLW2H.align.report.json --threads 32 --preset generic-amplicon-with-umi --save-output-file-names ./oneUMI/JLW2H/JLW2H.align.list --floating-left-alignment-boundary --floating-right-alignment-boundary J --rna --species hsa --tag-pattern ^(UMI:N{8})(R1:*)\^(R2:*) --assemble-clonotypes-by VDJRegion --impute-germline-on-export -M assemble.cloneAssemblerParameters.addReadsCountOnClustering=true ./oneUMI/JLW2H_R1A.fastq ./oneUMI/JLW2H_R2A.fastq ./oneUMI/JLW2H/JLW2H.vdjca
Analysis time: 3.47m
Total sequencing reads: 1231835
Successfully aligned reads: 1200236 (97.43%)
Coverage (percent of successfully aligned):
  CDR3: 1189531 (99.11%)
  FR3_TO_FR4: 1117808 (93.13%)
  CDR2_TO_FR4: 1117778 (93.13%)
  FR2_TO_FR4: 1116756 (93.04%)
  CDR1_TO_FR4: 1116359 (93.01%)
  VDJRegion: 1114776 (92.88%)
Alignment failed: no hits (not TCR/IG?): 3885 (0.32%)
Alignment failed after alignment-aided overlap: 12 (0%)
Alignment failed: absence of V hits: 6 (0%)
Alignment failed: absence of J hits: 26339 (2.14%)
Alignment failed: no target with both V and J alignments: 1357 (0.11%)
Overlapped: 1223011 (99.28%)
Overlapped and aligned: 1199080 (97.34%)
Overlapped and not aligned: 23931 (1.94%)
Alignment-aided overlaps, percent of overlapped and aligned: 707 (0.06%)
No CDR3 parts alignments, percent of successfully aligned: 105 (0.01%)
Partial aligned reads, percent of successfully aligned: 10600 (0.88%)
V gene chimeras: 386 (0.03%)
Paired-end alignment conflicts eliminated: 324 (0.03%)
Realigned with forced non-floating bound: 19062 (1.55%)
Realigned with forced non-floating right bound in left read: 311 (0.03%)
Realigned with forced non-floating left bound in right read: 311 (0.03%)
IGH chains: 1200071 (99.99%)
IGH non-functional: 17462 (1.46%)
IGK chains: 53 (0%)
IGK non-functional: 0 (0%)
IGL chains: 112 (0.01%)
IGL non-functional: 4 (3.57%)
Trimming report:
  R1 reads trimmed left: 1 (0%)
  R1 reads trimmed right: 15 (0%)
  Average R1 nucleotides trimmed left: 2.435391103516299E-6
  Average R1 nucleotides trimmed right: 6.859684941570908E-4
  R2 reads trimmed left: 230 (0.02%)
  R2 reads trimmed right: 89 (0.01%)
  Average R2 nucleotides trimmed left: 3.888507795281024E-4
  Average R2 nucleotides trimmed right: 0.0030101434039461455
Tag parsing report:
  Execution time: 0ns
  Total reads: 1231835
  Matched reads: 1231835 (100%)
  Projection +R1 +R2: 1231835 (100%)
  For variant 0:
    For projection +R1 +R2:
      UMI:Left position: 0
      R1:Left position: 8
      UMI:Right position: 8
      R1:Right position:
        186~249: + 4694 (0.38%) = 4694 (0.38%)
        250: + 1227141 (99.62%) = 1231835 (100%)
      R2:Left position: 0
      Variants: 0
      Cost: 0
      UMI length: 8
      R1 length:
        178~241: + 4694 (0.38%) = 4694 (0.38%)
        242: + 1227141 (99.62%) = 1231835 (100%)
      R2 length: 250
·····································································································
Analysis date: Sun May 26 16:19:13 CST 2024
Input file(s): ./oneUMI/JLW2H/JLW2H.vdjca
Output file(s): ./oneUMI/JLW2H/JLW2H.refined.vdjca
Version: 4.6.0; built=Sun Dec 10 03:48:42 CST 2023; rev=c9fafa41fe; lib=repseqio.v4.0
Command line arguments: refineTagsAndSort --report ./oneUMI/JLW2H/JLW2H.refine.report.txt --json-report ./oneUMI/JLW2H/JLW2H.refine.report.json ./oneUMI/JLW2H/JLW2H.vdjca ./oneUMI/JLW2H/JLW2H.refined.vdjca
Analysis time: 1.59m
Time spent in correction: 1.12m
Number of input records: 1200236
Number of output records: 1185174 (98.75%)
UMI correction report:
  UMI input diversity: 41676
  UMI output diversity: 33572 (80.55%)
  UMI input reads: 1200236
  UMI output reads: 1200234 (100%)
  UMI mean reads per tag: 28.8
  UMI input core diversity: 39869 (95.66%)
  UMI input core reads: 1198429 (99.85%)
  UMI directly corrected diversity: 8102 (19.44%)
  UMI directly corrected reads: 8953 (0.75%)
  UMI diversity filtered by tag quality: 2 (0%)
  UMI reads filtered by tag quality: 2 (0%)
  UMI diversity filtered by whitelist: 0 (0%)
  UMI recursively corrected: 724
Filter report:
  Number of groups: 33572
  Number of groups accepted: 28514 (84.93%)
  Total records weight: 1200234
  Records weight accepted: 1185174 (98.75%)
  Operator #0:
    Effective threshold: 6.0
    Nested thresholds:
      #0: 6
      #1: 26
·····································································································
Analysis date: Sun May 26 16:19:13 CST 2024
Input file(s): ./oneUMI/JLW2H/JLW2H.refined.vdjca
Output file(s): ./oneUMI/JLW2H/JLW2H.clns
Version: 4.6.0; built=Sun Dec 10 03:48:42 CST 2023; rev=c9fafa41fe; lib=repseqio.v4.0
Command line arguments: assemble --report ./oneUMI/JLW2H/JLW2H.assemble.report.txt --json-report ./oneUMI/JLW2H/JLW2H.assemble.report.json ./oneUMI/JLW2H/JLW2H.refined.vdjca ./oneUMI/JLW2H/JLW2H.clns
Analysis time: 42.54m
Final clonotype count: 7356
Reads used in clonotypes, percent of total: 340261 (27.62%)
Average number of reads per clonotype: 46.26
Reads dropped due to the lack of a clone sequence, percent of total: 84133 (6.83%)
Reads dropped due to a too short clonal sequence, percent of total: 0 (0%)
Reads dropped due to low quality, percent of total: 0 (0%)
Reads dropped due to failed mapping, percent of total: 36814 (2.99%)
Reads dropped with low quality clones, percent of total: 5178 (0.42%)
Aligned reads processed: 382253
Reads used in clonotypes before clustering, percent of total: 340261 (27.62%)
Number of reads used as a core, percent of used: 338087 (99.36%)
Mapped low quality reads, percent of used: 2174 (0.64%)
Reads clustered in PCR error correction, percent of used: 1378 (0.4%)
Reads pre-clustered due to the similar VJC-lists, percent of used: 0 (0%)
Clonotypes dropped as low quality: 298
Clonotypes eliminated by PCR error correction: 61
Clonotypes pre-clustered due to the similar VJC-lists: 0
Clones dropped in post filtering: 0 (0%)
Reads dropped in post filtering: 0.0 (0%)
Alignments filtered by tag prefix: 0 (0%)
IGH chains: 7356 (100%)
IGH non-functional: 130 (1.77%)
Pre-clone assembler report:
  Number of input groups: 28514
  Number of input groups with no assembling feature: 38
  Number of input alignments: 1185174
  Number of alignments with assembling feature: 1101041 (92.9%)
  Number of output pre-clones: 20474
  Number of pre-clonotypes per group:  
    0: + 9494 (33.34%) = 9494 (33.34%)
    1: + 17525 (61.54%) = 27019 (94.88%)
    2: + 1422 (4.99%) = 28441 (99.88%)
    3: + 35 (0.12%) = 28476 (100%)
  Number of assembling feature sequences in groups with zero pre-clonotypes: 467190
  Number of dropped pre-clones by tag suffix conflict: 0
  Number of dropped alignments by tag suffix conflict: 0
  Number of core alignments: 379286 (32%)
  Discarded core alignments: 721755 (190.29%)
  Empirically assigned alignments: 2967 (0.25%)
  Empirical assignment conflicts: 86 (0.01%)
  Tag+VJ-gene empirically assigned alignments: 3053 (0.26%)
  VJ-gene empirically assigned alignments: 0 (0%)
  Tag empirically assigned alignments: 0 (0%)
  Number of ambiguous groups: 1457
  Number of ambiguous V-genes: 321
  Number of ambiguous J-genes: 595
  Number of ambiguous tag+V/J-gene combinations: 916
  Ignored non-productive alignments: 0 (0%)
  Unassigned alignments: 802470 (67.71%)
·····································································································

**Comment by mizraelson** [comment, marked as answer]
One of the major reasons for that could be a short UMI sequence.
`0: + 9494 (33.34%) = 9494 (33.34%)`
For 9494 UMIs, no consensus was assembled, possibly because there were multiple clones marked with the same UMI. For example, if a UMI is covered by 10 reads, but each read has a different CDR3, no consensus will be assembled.

**Reply by liyunjian** [reply]
thank you for your reply.
